sysfs/net_class.go

```diff
@@ -113,71 +137,140 @@ func (nc NetClass) parseNetClassIface(devicePath string) (*NetClassIface, error)
 	}
 
 	for _, f := range files {
-		if !f.Mode().IsRegular() {
+		if !f.Mode().IsRegular() && f.Name() != "statistics" {
 			continue
 		}
-		name := filepath.Join(devicePath, f.Name())
-		value, err := util.SysReadFile(name)
-		if err != nil {
-			if os.IsNotExist(err) || os.IsPermission(err) || err.Error() == "operation not supported" || err.Error() == "invalid argument" {
-				continue
+
+		if f.Name() == "statistics" {
```

Comment:
I think it would be better to have a separate parseNetClassIfaceStatistics function that is invoked in NetClass, unless there are reasons to do it this way. @pgier @SuperQ thoughts?

Reply:
Yes, I think it would be a bit more readable to push the statistics code down in to a function.  I added some pseudo-code above with one idea.  @chris-parker-za WDYT?